src/stores/AutoRageshakeStore.ts

```diff
@@ -113,6 +126,12 @@ export default class AutoRageshakeStore extends AsyncStoreWithClient<IState> {
         }
     }
 
+    private async onSyncStateChange(_state: SyncState, _prevState: SyncState, data: ISyncStateData) {
+        if (!this.state.initialSyncCompleted) {
+            await this.updateState({ initialSyncCompleted: !!data.nextSyncToken });
+        }
```

Comment:
Does it need to specifically be the initial sync or the first non-cached sync? `ReadyWatchingStore` is an example of how we check for the first non-cached sync, fwiw.